pkg/arvo/neo/cod/std/src/fil/diary.md

```diff
@@ -0,0 +1,297 @@
+^-  kook:neo
+|%
+++  state  only/%diary
+++  poke   (sy %diary-diff ~)
+++  kids   `y/(~(gas by *lads:neo) ~[[[|/%da |] [only/%txt ~]]])
+++  deps   *deps:neo
+++  form
+  ^-  form:neo
+  |_  [=bowl:neo =aeon:neo =pail:neo]
+  ++  init
+    |=  old=(unit pail:neo)
+    ^-  (quip card:neo pail:neo)
+    [~ ?~(old diary/!>(*diary) u.old)]
+  ++  poke
+    |=  [=stud:neo vax=vase]
+    ^-  (quip card:neo pail:neo)
+    ?>  =(%diary-diff stud)
+    =/  state  !<(diary q.pail)
+    =/  act    !<(diary-diff vax)
+    ?>  =(our ship.src):bowl
+    ?-    -.act
+       %put-entry
+      :_  diary/!>(state)
+      ~[[(welp here.bowl ~[da/id.act]) [%make %txt `txt/!>(txt.act) ~]]]
+        %del-entry
+      :_  diary/!>(state)
+      ~[[(welp here.bowl ~[da/id.act]) [%tomb ~]]]
+    ==
+  --
+--
+```
+
+Most of this should be legible after the first tutorial. The only new ideas are the `+kids` arm, the use of `bowl`, and `%tomb`.
+
+## +kids
+Every shrub is a node in one tree, with one “root shrub” at the top of that tree. Every shrub below the root shrub is either one of its immediate children or one of its descendants.
+
+In its `+kids` arm, every shrub can define constraints for the shrubs below it in the namespace, whether that’s constraining the types of their state or the pokes they’ll accept from other shrubs.
+
+Let’s expand on diary’s `+kids` arm.
+
+```hoon
+::
+::  constrain shrubs below diary in the namespace
+::  by defining the types of their state and pokes
+++  kids
+  ::
+  ::  kids:neo is a (unit port:neo)
+  ^-  kids:neo
+  %-  some
+  ::
+  ::  port:neo is (pair dare:neo lads:neo)
+  ::  dare:neo is ?(%y %z)
+  ::  if %y, only constrain our immediate children
+  ::  if %z, recursively constrain all descendants
+  :-  %y
+  ::  lads:neo is (map pish:neo lash:neo)
+  %-  ~(gas by *lads:neo)
+  :~  :-  ::
+          ::  pish:neo
+          ::  to simplify: [%.n @da] means the kid's
+          ::  path contains any @da, and %.n is there
+          ::  to signify that the pith can not have more
+          ::  fields afterwards
+          [[%.n %da] %.n]
+      ::
+      ::  lash:neo is (pair curb:neo (set stud:neo))
+      ::  curb:neo defines the kids' state
+      ::  (set stud:neo) defines the kids' pokes
+      [[%only %txt] ~]
+  ==
+```
+
+In this case, Diary only constrains its immediate children. Each of its children lives at `/path/to/diary/<@da>`, where `<@da>` is used as an ID for the entry, and those children are just `%txt`s (cords) which store the text of the diary entry.
+
+There’s no need for Diary to recusrively constrain every single path that nests under `/path/to/diary`, since the app only needs to reserve the first “generation” of shrubs below it in order to work. In this case, the developer has forbidden any shrubs being created under the diary entries by ending the `pish` in a `%.n`. If that were `%.y`, any app could make a shrub at `/path/to/diary/<@da>/comments`. or `/path/to/diary/<@da>/backlinks`.
```

Comment:
Would need to do more than just change `%.n` to `%.y`, would also need to change `%y` to `%z`.